articles/architecture-scenarios/b2c/authentication.md

```diff
@@ -0,0 +1,120 @@
+
+::: note
+Anomaly Detection is handled by Auth0 and provides a great security feature for your product. If you are going to utilize it, ensure that you have set up your SMTP provider and configured your email templates before turning on email delivery to your users.
+:::
+
+## Application Integration
+
+Once you have figured out how you want to authenticate your users, the next step is to determine how and when you will initiate that authentication. Each application may have its own starting point.
+
+::: warning
+Native Mobile Applications (and desktop applications) should use the system browser for authentication, or they open themselves up to additional security risks.
+:::
+
+As discussed, OpenID Connect (OIDC) is the most frequently used industry standard protocol when it comes to customer facing applications. Figuring out which OIDC flow to use is your first task, and you will want to start by reviewing guidance concerning the various grant types here.  
+
+If you want to allow anonymous users access to any part of our application then you need to determine if you will be redirecting right away or prompting your users to redirect only when required (or perhaps some combination of both; see here for further discussion). If users can deep link to a protected version (or area) of your site then you will need to determine the links to your application that will result in an automatic redirect to Auth0 (see here for further information). 
+
+### Anonymous Access
+
+It is important to consider the user experience when someone first comes to the application. If your application supports anonymous user access (this is quite common for eCommerce applications). There are different scenarios within this context:
+
+* They are returning to the application after having already logged in, or
+* This is the first time they are accessing the application:
+    * They have already accessed a different application that uses the same Auth0 tenant,
+    * They have never (or not in a long time) authenticated on this device or browser.
+
+When an anonymous user accesses your application, it can often be desirable for the application to discover if the user has already logged into a different application in the same family, or to remember this user even if the application is a SPA with no state. For example, if you can determine that the user is already logged in, you might decide to have the header of the application skip displaying a login button and instead have an account or profile menu for the user. To accomplish this you will want to utilize "silent authentication". Silent authentication will allow you to check to see if the user is logged in without prompting them to log in if they are not. Then the application can present a login button if necessary. If the user is logged in already, however, then you will receive tokens and will not have to present the user with a login button again.
```

Comment:
```suggestion
When an anonymous user accesses your application it can often be desirable for the application to discover if the user has already logged into a different application in the same family, or to remember this user even if the application is a SPA with no state. For example, if you can determine that the user is already logged in you might decide to have the header of the application skip displaying a login button and instead have an account or profile menu for the user. To accomplish this you will want to utilize "silent authentication". Silent authentication will allow you to check to see if the user is logged in without prompting them to log in if they are not. Then the application can present a login button if necessary. If the user is logged in already, however, then you will receive tokens and will not have to present the user with a login button again.
```